src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1056,6 +1094,12 @@ protected internal struct EventData
             /// </summary>
             public int Size { get { return m_Size; } set { m_Size = value; } }
 
+            /// <summary>
+            /// Reserved by ETW.  This property is present to ensure that we can zero it
+            /// since System.Private.CoreLib uses are not zero'd.
+            /// </summary>
+            internal int Reserved { get { return m_Reserved; } set { m_Reserved = value; } }
```

Comment:
What will the recommended pattern be for code outside of corelib that decides to enable clearing of initlocals?
```C#
descrs[0] = new EventData
{
    DataPointer = ...,
    Size = ...
};
```
instead of:
```C#
descrs[0].DataPointer = ...;
descrs[0].Size = ...;
```
?

Reply:
`EventData` is opaque structure with no public fields. Anybody using today it has to create their own local definition with matching layout to be able to write into it.

Reply:
That's a good question.  I was not aware of this effort until recently, and did not know that this is something that will go beyond CoreLib.  Can you tell me more about how this will work or point me to something that I can read?

It's possible to pre-initialize the structure prior to using it, but that might be a big hammer:

```
EventData *pData = stackalloc EventData[4];
for(int i=0; i<4; i++)
{
    pData[i] = default(EventData);
}
```

We can also consider adding a public property to EventData, but I feel like I need to understand more about the overall goal of allowing people to clear preinitlocals and what if anything else besides stackalloc is affected by this flag.

Reply:
It took me a minute to catch on to your recommendation above.  Your proposal would also work:

```
descrs[0] = new EventData
{
    DataPointer = ...,
    Size = ...
};
```

Reply:
> Can you tell me more about how this will work or point me to something that I can read?

There's an ILLink step that can clear the flag:
https://github.com/mono/linker/blob/86b09e1033d68d4cc1aaa2c7362c16b756d6cbf3/corebuild/integration/ILLink.CustomSteps/ClearInitLocals.cs

There's also a proposal to support this directly in Roslyn:
https://github.com/dotnet/csharplang/blob/master/proposals/skip-localsinit.md

The latter link has a good description of the issues and affect.

Reply:
Thanks @stephentoub.  I will follow-up on this to see if we want to do anything more than:

```
descrs[0] = new EventData
{
    DataPointer = ...,
    Size = ...
};
```

Reply:
@stephentoub' suggestion works (although I have not looked at its codegen, it may be good).  

I debated making DataPointer clear the Reserved field, but that penalizes the common case for the sake of the uncommon case.   

I think that advice is fine.